app/components/ChangeColorInput.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component, PropTypes } from 'react';
+import BaseComponent from './BaseComponent';
+import { CompactPicker } from 'react-color';
+
+export default class ChangeColorInput extends BaseComponent {
+
+  constructor() {
+    super();
+    this.state = {
+      displayColorPicker: false,
+      color: "#cccccc"
+ 	 },
+    this.handleClick = this.handleClick.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleValueChange = this.handleValueChange.bind(this);
+    this.handleClearClick = this.handleClearClick.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ displayColorPicker: !this.state.displayColorPicker });
+  }
+
+  handleClose() {
+    this.setState({ displayColorPicker: false });
+  }
+
+  handleClearClick() {
+  	this.handleClose();
+  	this.onChange("#cccccc");
+  }
+
+  handleValueChange(color) {
+    this.onChange("#" + color.hex);
+  }
+
+  onChange(newColor) {
+  	this.setState({ color: newColor });
+    this.props.parent.apply(newColor);
```

Comment:
instead of passing itself as a `parent` prop, the parent component should pass `apply()` as `onChange()` or something like that. so this line would become `this.props.onChange(newColor)`.